lib/index.js

```diff
@@ -155,12 +155,23 @@ internals.handler = function (route, handlerOptions) {
             };
 
             // Encoding request path
-            const _encodeURL = (url) => {
-                if (decodeURI(url) === url) {
-                    url = encodeURI(url);
+            const _encodeURL = (uri) => {
+                let uriBasePath = uri;
+                let haveQueryParam = false;
+                let queryParams = '';
+
+                if (uri.match(/\?./)) {
```

Comment:
Have a look at this commit 
We will likely merge it soon 
I used URL nodejs package to split the url into parts 
https://github.com/sirensolutions/h2o2/commit/82714624ffe3ae6623b01a3c5d00443a21614288

```
const URL = require('url');
```
It is best to use this native package then to write a custom code for parsing urls 
Could you also point me to the issue where the original bug was described ?

@johnnylopes  Maybe we should think about integration test for ingestion which would cover the scenario from the original ticket (ask @sibisimon) 
and another one to cover the investigate dev tools - as the fix done initially broke them (but only in dev mode so I do not know how we could test that)

Reply:
https://sirensolutions.atlassian.net/browse/INVE-274